### DISCUSSION TITLE
PNG files instead of fontawesome icons

### DISCUSSION THREAD

**Discussion by zp1956** [OpenHantek/OpenHantek6022#378] [Q&A] [answered]
I changed the code to use PNG images (32x32) as icons instead of fontawsome icons. Compiling in windows environment passes.

Then I tried to delete the corresponding #include expressions, as well as the complete src/iconfont and res/fonts folders, but the compiler reports errors and stops.

Is there an easy way to delete everything related to fontawesome from the code as well as delete files that are no longer needed?

**Comment by Ho-Ro** [comment]
Interesting idea, do you have a GH fork with these changes? This would help me to understand and support your efforts - unfortunately I'm currently away from my development system, so it may take some weeks on my side. Nevertheless these icons are legacy from the old openhantek project and I used them without thinking of alternatives.

**Comment by zp1956** [comment]
thank you for your reply
no, I haven't created a GH fork but I plan to set it up soon.

**Comment by zp1956** [comment]
[GH fork](https://github.com/zp1956/OpenHantek6022) is created. In the commit #1, the fontawesome icons were replaced with PNG files and the code was changed accordingly. In the commit #2, unnecessary files were deleted and several statements were disabled (commented).

Environment:
- Windows 10
- Qt 5.8.0
- cmake 3.26.3
- SainSmart DDS120


EDIT Ho-Ro 20230826: link to GH fork added above.

**Comment by Ho-Ro** [comment, marked as answer]
Please check my latest commit 0806fe6 where all traces of FA are purged. I replaced the icons with B/W icons to be consistent with the icons visible in the tools bar on the left.
To fix the error messages you need to add this line to `mainwindow.cpp`.
```c
#include <QPainter>
```

**Comment by zp1956** [comment]
Thank you. You confirmed my goal. SVG or PNG is a matter of personal choice.
I hope this will be part of the next version. I see no reason to use FA and associated files for those 10 or so icons.

**Reply by Ho-Ro** [reply]
> SVG or PNG is a matter of personal choice.

I prefer to edit SVGs rather than PNGs.

> I hope this will be part of the next version.

TL; DR: Yes!

My development model works on the `main` trunk; after a stable release (e.g. 3.3.3), development continues in `main` until the next release is ready. The `unstable` releases serve as development drops, where *unstable* does not mean that the program crashes, but that new features have not yet reached their final form and also the GUI can still change. Only in rare cases of major restructuring do I create an extra feature branch on GitHub. Of course I branch in my local repository, but then I merge it before putting it online.

**Comment by zp1956** [comment]
Ok.
Don't forget to remove QtAwesomeFree.qrc and fontawesome-4.7.0.ttf from source.

Please check tooltips definitions. I have disabled that part and left yellow background.

Main.cpp
	//openHantekApplication.setStyleSheet( "QToolTip { border: 2px solid white; padding: 2px; border-radius: 5px; font-weight: bold; "
	//                                     "color: white; background-color: black; }" );

Sispinbox.cpp
	void SiSpinBox::setBackground() {
	//QPalette palette;
	//setStyleSheet( "SiSpinBox {color: " + palette.color( QPalette::Text ).name() +
	//               "; background-color: " + palette.color( QPalette::Mid ).name() + " }" );
	//setStyleSheet( "QToolTip { color: " + palette.color( QPalette::ToolTipText ).name() +
	//               "; background-color: " + palette.color( QPalette::ToolTipBase ).name() + " }" );
	}

Otherwise I get this
![image](https://github.com/OpenHantek/OpenHantek6022/assets/27331794/11ff94c3-d554-4b8f-a2a9-8e99be7eb6f8)
